### DISCUSSION TITLE
Adding Z-Normalization before or after RandomNoise

### DISCUSSION THREAD

**Discussion by nicoloesch** [TorchIO-project/torchio#843] [Q&A] [answered]
Hey,

I have a question regarding the Z-Normalization and its position in the pipeline w.r.t RandomNoise. According to the online Colab, the Z-Norm should be applied before adding RandomNoise. However, if we normalize the image to zero mean and unit variance and add noise with varying mean/variance, wouldn't that alter our mean and variance of the image again, making the Z-Norm "obsolete"?

Thank you for your help :)

**Comment by romainVala** [comment]
Hello
yes you are right you won't have zscore any more
which Colab are you refering to ?

**Reply by nicoloesch** [reply]
Hey,

It is in the [Data and Augmentation Tutorial with a link to the Random Noise cell](https://colab.research.google.com/github/fepegar/torchio-notebooks/blob/main/notebooks/Data_preprocessing_and_augmentation_using_TorchIO_a_tutorial.ipynb#scrollTo=KRaLonXa8YzE).

If the above link to the specific cell is not working for you, use the [link to the regular tutorial](https://colab.research.google.com/github/fepegar/torchio-notebooks/blob/main/notebooks/Data_preprocessing_and_augmentation_using_TorchIO_a_tutorial.ipynb) , navigate to **Intensity Transforms > Augmentation > Random Noise**.
There is written: 

> Gaussian noise can be simulated using [RandomNoise](https://torchio.readthedocs.io/transforms/augmentation.html#randomnoise). This transform is easiest to use after [ZNormalization](https://torchio.readthedocs.io/transforms/preprocessing.html#torchio.transforms.ZNormalization), as we know beforehand that the mean and standard deviation of the input will be 0 and 1, respectively. If necessary, the noise mean and std can be set using the corresponding keyword arguments.

I am just wondering about the sequence of operations as RandomNoise **before** Z-Normalization would make more sense to me - unless one adds noise with zero mean and unit variance if used **after** the normalization (which however would make the parameters obsolete as only one configuration can be used)

Thank you for your help - maybe I am confusing some things here :)

**Comment by romainVala** [comment, marked as answer]
As explain the point here is that adding noise to an image it is difficult to choose a "relevant"  std for the noise, 
(imagine you have intensity mean around 200 in one subject and 2000 in an other (that can append for MRI because it is always define with an arbitrary scale factor)
so adding a noise with a std of 10 will have very different effect of those 2 subjects

that is why "normalising the intesity" first is important. to be able to keep a given std (for RandomNoise).  Note that you could also do it with the `RescaleIntensity` transform

if Z-Normalization is important, you can just add it a second time after adding random noise
